Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,265 @@
+/-
+Copyright (c) 2023 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+
+import Mathlib.Algebra.Algebra.Unitization
+import Mathlib.Analysis.NormedSpace.OperatorNorm
+
+/-!
+# Unitization norms
+
+Given a not necessarily unital normed `𝕜`-algebra `A`, it is frequently of interest to equip its
+`Unitization` with a norm which simultaneously makes it into a normed algebra and also satisfies
+two properties:
+
+- `‖1‖ = 1` (i.e., `NormOneClass`)
+- The embedding of `A` in `Unitization 𝕜 A` is an isometry. (i.e., `Isometry Unitization.inr`)
+
+One way to do this is to equip it with the norm from `PiLp 1` (actually, it should be
+`ProdLp 1`, but that doesn't exist), that is, `‖(k, a)‖ = ‖k‖ + ‖a‖`. However, when the norm on `A`
+is *regular* (i.e., `ContinuousLinearMap.mul`) is an isometry, there is another natural choice:
+the pullback of the norm on `𝕜 × (A →L[𝕜] A)` under the map
+`(k, a) ↦ (k, k • 1 + ContinuousLinearMap.mul 𝕜 A a)`. It turns out that among all norms on the
+unitization satisfying the properties specified above, the norm inherited from `PiLp 1` is maximal,
+and the norm inherited from this pullback is minimal.
+
+For possibly non-unital `RegularNormedAlgebra`s  `A` (over `𝕜`), we construct a `NormedAlgebra`
+structure on `Unitization 𝕜 A` using the pullback described above. The reason for choosing this norm
+is that for a C⋆-algebra `A` its norm is always regular, and the pullback norm on `Unitization 𝕜 A`
+is then also a C⋆-norm.
+
+## Main definitions
+
+- `NonUnitalAlgHom.Lmul : A →ₙₐ[𝕜] A →L[𝕜] A`: `ContinuousLinearMap.mul` upgraded to a non-unital
+  algebra homomorphism. This is the left regular representation of `A`.
+- `Unitization.splitMul : Unitization 𝕜 A →ₐ[𝕜] (𝕜 × (A →L[𝕜] A))`: The first coordinate of this
+  map is just `Unitization.fst` and the second is the `Unitization.lift` of the left regular
+  representation of `A` (i.e., `NonUnitalAlgHom.Lmul`). We use this map to pull back the
+  `NormedRing` and `NormedAlgebra` structures.
+
+## Implementation details
+
+We ensure that the uniform structure, and hence also the topological structure, is definitionally
+equal to `instUniformSpaceProd` (viewing `Unitization 𝕜 A` as `𝕜 × A`) by means of forgetful
+inheritance. The same is true of the bornology.
+
+-/
+
+variable (𝕜 A : Type _) [NontriviallyNormedField 𝕜] [NonUnitalNormedRing A]
+variable [NormedSpace 𝕜 A] [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A]
+
+open ContinuousLinearMap
+
+/-- Multiplication on the left in a non-unital algebra `A` as a non-unital algebra homomorphism
+into the algebra of *continuous* linear maps. This has more algebraic structure than
+`ContinuousLinearMap.mul`, but there is no longer continuity bundled in the first coordinate. -/
+noncomputable def NonUnitalAlgHom.Lmul : A →ₙₐ[𝕜] A →L[𝕜] A :=
```

Comment:
This should probably move to an earlier file. Also I'm not sure about the name, does `lmul` stand for "left multiplication" or "linear multiplication"?

Reply:
I believe it means "linear" because [ContinuousLinearMap.mulLeftRight](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/NormedSpace/OperatorNorm.html#ContinuousLinearMap.mulLeftRight) used to be called `continuous_linear_map.lmul_left_right`.

Reply:
moved to `OperatorNorm` file, immediately after `ContinuousLinearMap.mul`